File: backend/src/main/java/com/eet/backend/model/ExchangeRate.java
package com.eet.backend.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "exchange_rates", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"from_currency", "to_currency", "date"})
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ExchangeRate {

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(name = "rate_id", updatable = false, nullable = false)
    private UUID rateId;

    @Column(name = "from_currency", nullable = false, length = 3)
    private String fromCurrency;

    @Column(name = "to_currency", nullable = false, length = 3)
    private String toCurrency;

    @Column(nullable = false, precision = 19, scale = 6)
    private BigDecimal rate;

    @Column(nullable = false)
    private LocalDate date;

    public BigDecimal convert(BigDecimal amount) {
        return amount.multiply(rate);
    }
}

File: frontend/ErasmusExpenseTracker-front/src/models/ExchangeRate.js
export default class ExchangeRate {
  constructor({ rateId, fromCurrency, toCurrency, rate, date }) {
    this.rateId = rateId;
    this.fromCurrency = fromCurrency;
    this.toCurrency = toCurrency;
    this.rate = parseFloat(rate);
    this.date = new Date(date); // LocalDate desde backend (yyyy-MM-dd)
  }

  isExpired(days = 1) {
    const now = new Date();
    const diffMs = now - this.date;
    return diffMs > days * 24 * 60 * 60 * 1000;
  }

  formattedRate() {
    return `${this.fromCurrency} → ${this.toCurrency}: ${this.rate.toFixed(4)}`;
  }
}
